### ISSUE TITLE
Docker container runs while netdata process is gone

### ISSUE THREAD

**Issue by jakubgs** [netdata/netdata#4493]
##### Bug report summary
The `netdata` service stops responding despite and the process is gone but the container keeps running.

##### OS / Environment
```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic
```
Running inside of docker:
```
Docker version 18.06.1-ce, build e68fc7a
```

##### Steps To Reproduce
I have no idea. It just randomly stops working.

```yaml
[global]
	hostname = node-01.do-ams3.les-rops.misc
	run as user = root
	update every = 5
	memory mode = ram
	# debug log = /var/log/netdata/debug.log
	# error log = /var/log/netdata/error.log
	access log = /dev/null

[health]
	enabled = no

[statsd]
	enabled = no

[plugins]
	# enable running new plugins = yes
	# check for new plugins every = 60
	idlejitter = no
	python.d = no
	node.d = no
	charts.d = no
	fping = no

[plugin:apps]
	update every = 10

[plugin:proc:diskspace]
	update every = 10
	check for new mount points every = 0

[plugin:proc]
	/proc/net/snmp = no
	/proc/net/snmp6 = no
```

##### Expected behavior
I expect the docker container to at least stop when the process fails.

##### Logs
The logs show nothing, they just stop at some point.

##### Details
I can see the container is running:
```bash
 % docker ps
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS                                                                     NAMES
ecffc0067756        titpetric/netdata:1.10       "/run.sh"                2 weeks ago         Up 2 days           0.0.0.0:8001->19999/tcp                                                   netdata
```
I can verify that with `inspect`
```bash
 % docker inspect netdata | grep running 
            "Status": "running",
```
But when I check the processes running it's empty:
```bash
 % docker top netdata
UID                 PID                 PPID                C                   STIME               TTY                 TIME                CMD
```
And when I try to `exec` a shell in the container to access it I get:
```bash
 % docker exec -it netdata bash        
cannot exec in a stopped state: unknown
```
This has happened several times now and every time there are no errors in the logs and the container just keeps running. This does not happen with any other containers I have.

**Comment by ktsaou** [comment]
@jakubgs keep in mind we supply docker images as `netdata/netdata`. You may want to give the official one a try.

**Comment by jakubgs** [comment]
Oh, I had no idea, because you link both in the README of the repo:
https://github.com/netdata/netdata#user-base

* https://hub.docker.com/r/titpetric/netdata/
* https://hub.docker.com/r/netdata/netdata/

But that's good to know. I'll try the other one and get back to you.

**Comment by ktsaou** [comment]
That is ok. @titpetric	does a marvelous job providing containers for netdata too.

**Comment by paulfantom** [comment]
Strangely I cannot reproduce this. Killing a netdata process kills a container when using titpetric images. Same goes for official `netdata/netdata` ones.

Something like this should be impossible when running `netdata/netdata` docker containers as netdata process is running with PID 1 inside a container. 

```
[paulfantom@nas ~]$ docker ps | grep netdata
4261b17e305f        netdata/netdata:latest                    "/usr/sbin/run.sh"       16 hours ago        Up 16 hours            0.0.0.0:19999->19999/tcp           monitoring_netdata_1
[paulfantom@nas ~]$ docker exec -it monitoring_netdata_1 ps aux
PID   USER     TIME  COMMAND
    1 netdata  20:05 /usr/sbin/netdata -u netdata -D -s /host -p 19999
   23 netdata   7:11 /usr/bin/python /usr/libexec/netdata/plugins.d/python.d.pl
  675 root      0:25 /usr/libexec/netdata/plugins.d/apps.plugin 1
  691 root      0:00 ps aux
```

We should probably uncomment this line to be more resilient: https://github.com/netdata/netdata/blob/c6cce6bd2accec27a61243e1ffa38826ac2a4892/docker/run.sh#L3


That said, it is entirely possible that something in titpetric [`run.sh` script](https://github.com/titpetric/netdata/blob/master/releases/v1.10.0/scripts/run.sh) can cause such problem in some environments. [Ours](https://github.com/netdata/netdata/blob/master/docker/run.sh) shouldn't be affected.


@jakubgs could you check if you have same problems when running `netdata/netdata`?

Marking this as a `wontfix` as an issue currently relates not to official container images and not to current version.

**Comment by jakubgs** [comment]
Yeah, I've tested this with `netdata/netdata:latest-amd64` and it seems like now it properly crashes the container and I can see it's state changing to `Restarting (137) 43 minutes ago` rather than being stuck in libo, which is an improvement of sorts.

I will try enabling some debugging options in order to track down what the issue is.

**Comment by jakubgs** [comment]
I did get something in the logs, not sure if it's relevant to the crash:
```
netdata ERROR : PLUGIN[cgroups] : heartbeat missed 4805115 monotonic microseconds
netdata ERROR : PLUGINSD[apps] : read failed
netdata ERROR : PLUGINSD[apps] : '/usr/libexec/netdata/plugins.d/apps.plugin' (pid 196) disconnected after 36465 successful data collections (ENDs). (errno 22, Invalid argument)
netdata ERROR : PLUGINSD[apps] : child pid 196 killed by signal 9.
netdata ERROR : PLUGINSD[apps] : '/usr/libexec/netdata/plugins.d/apps.plugin' (pid 196) exited with error code -1, but has given useful output in the past (670505 times). Waiting a bit before starting it again. (errno 22, Invalid argument)
netdata ERROR : PLUGIN[proc] : heartbeat missed 10015738 monotonic microseconds
```

**Comment by titpetric** [comment]
I think this may be a Docker and/or kernel bug, not necessarily a packaging one.

- https://github.com/moby/moby/issues/37009

I suggest closing as it's not likely we can do anything but wait for official patches. I'd suggest upgrading your kernel if it's older than 3.10, but beyond that, just track the upstream issue for info.

**Comment by jakubgs** [comment]
Well, this is still happening, the container process stops but not the container itself:
```bash
 % d ps 
CONTAINER ID        IMAGE                          COMMAND                  CREATED             STATUS              PORTS                      NAMES
006bf4b87b11        netdata/netdata:latest-amd64   "/usr/sbin/run.sh"       25 hours ago        Up 20 hours         0.0.0.0:8001->19999/tcp    netdata
 % ps faux
...(ommited)...
root     26524  0.4  1.1 1248832 44812 ?       Ssl  Nov08   5:14 /usr/bin/dockerd -H fd://
root     26547  0.6  0.4 1099004 19056 ?       Ssl  Nov08   7:20  \_ docker-containerd --config /var/run/docker/containerd/containerd.toml
root     26875  0.0  0.0   7564  1904 ?        Sl   Nov08   0:02  |   \_ docker-containerd-shim -namespace moby -workdir /var/lib/docker/containerd/daemon/io.containerd.runtime.v1.linux/moby/006bf4b87b117ea2e949bd3894fa2e5178b2055d5b6d0407289c46aa4dee759d -address /var/run/docker/containerd/docker-containerd.sock -containerd-binary /usr/bin/docker-containerd -runtime-root /var/run/docker/runtime-runc
...(ommited)...
 % docker exec -it netdata sh  
cannot exec in a stopped state: unknown
 % docker restart netdata
netdata
 % docker ps --filter='Name=netdata'
CONTAINER ID        IMAGE                          COMMAND              CREATED             STATUS              PORTS                     NAMES
006bf4b87b11        netdata/netdata:latest-amd64   "/usr/sbin/run.sh"   25 hours ago        Up 20 hours         0.0.0.0:8001->19999/tcp   netdata
 % sudo strace -p 26875              
strace: Process 26875 attached
futex(0x7fd008, FUTEX_WAIT, 0, NULL
```
The `docker-containerd-shim` process is running with nothing under it.
